skopt/searchcv.py

```diff
@@ -673,35 +555,47 @@ def step(self, X, y, space_id, groups=None, n_jobs=1):
         # optimizer minimizes objective, hence provide negative score
         optimizer.tell(params, [-score for score in local_results])
 
-        # fit the best model if necessary
-        if self.refit:
-            self._fit_best_model(X, y)
-
     def fit(self, X, y=None, groups=None):
         """Run fit on the estimator with randomly drawn parameters.
 
         Parameters
         ----------
-        X : array-like, shape = [n_samples, n_features]
-            Training vector, where n_samples in the number of samples and
-            n_features is the number of features.
+        X : array-like or sparse matrix, shape = [n_samples, n_features]
+            The training input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csc_matrix``.
 
         y : array-like, shape = [n_samples] or [n_samples, n_output]
-            Target relative to X for classification or regression;
+            Target relative to X for classification or regression (class
+            labels, as integers or strings).
 
         groups : array-like, with shape (n_samples,), optional
             Group labels for the samples used while splitting the dataset into
             train/test set.
         """
 
-        # check if the list of parameter spaces is provided. If not, then
-        # only step in manual mode can be used.
+        # check if space is a single dict, convert to list if so
+        search_spaces = self.search_spaces
+        if isinstance(search_spaces, dict):
+            search_spaces = [search_spaces]
 
-        if len(self.search_spaces_) == 0:
-            raise ValueError(
-                "Please provide search space using `add_spaces` first before"
-                "calling fit method."
-            )
+        self.search_spaces_ = {}
+        for space, name in zip(search_spaces, list(range(len(search_spaces)))):
+            self.search_spaces_[name] = space
+
+        self.optimizer_kwargs_ = {} if self.optimizer_kwargs is None else\
+            self.optimizer_kwargs
```

Comment:
I'd make this into:
```python
if self.optimizer_kwargs is None:
  self.optimizer_kwargs_ = {}
else:
  self.optimizer_kwargs_ = self.optimizer_kwargs
```

instead of trying to fit it on one line, then having to use `\` to continue on a second line :)